### ISSUE TITLE
Allow reading JSON strings as char[]

### ISSUE THREAD

**Issue by bannmann** [zalando-incubator/flatjson#3]
In many REST APIs, login & signup operations accept JSON objects with username and password properties. JSON parsers, however, always seem to create String instances for any JSON string property - even when you use an object mapping facility to map it to a `char[]` field.

This, of course, goes against the long-standing principle of using `char[]` for passwords so you can wipe the contents to make sure they don't linger in memory longer than needed (e.g. see all common password hasher APIs).

flatjson's approach of using an index and only allocating objects when neccessary seems uniquely suited to offer an alternative: if my code knows it's reading sensitive data, it could call an `asCharArray()` method instead of `asString()`.

Is this something you could and would implement?

**Comment by bannmann** [comment]
I implemented this as proof of concept and it indeed turned out to be feasible. However, doing this properly resulted in significant refactoring so I decided it's better fork the project. This way, the resulting library can also drop some features which are not required for its use case.

If anyone interested in the topic finds this: my fork is called [WhisperJson](https://github.com/bannmann/whisperjson/).

As the flatjson project will be less likely now to implement this as well, I'm closing the issue.

P.S.: Thank you, @tlossen, for your great work that I could build upon!
